### PR TITLE
COG-1369 Exact match search results

### PR DESCRIPTION
**Pull request by RamuKammagani-osi**
#### Which User story does this relate to (link)?
-  https://osi-cwds.atlassian.net/browse/COG-1369

#### Brief feature description
- returns the exact match search results based on search criteria.

### CODE REVIEW

File: app/javascript/_utils/formatters.test.js [hunk 1]
```diff
@@ -221,3 +222,40 @@ describe('#formatAdminWithRole', () => {
     expect(formatAdminWithRole(adminName, adminRole)).toEqual('Some Admin')
   })
 })
+
+// describe('#formatOffices', () => {
```
Comment (kelfink) [inline app/javascript/_utils/formatters.test.js]: Looks like we can remove this old test

File: app/javascript/views/searchUserList/SearchUserList.jsx [hunk 2]
```diff
@@ -115,56 +99,26 @@ class SearchUserList extends PureComponent {
...
-        />
+  renderExactMatches = () => {
+    const { exactMatches, officesList, rolesList, fetching } = this.props
+    return fetching ? (
+      <div style={{ textAlign: 'center', marginTop: '10%' }}>
+        <Icon name="circle-notch" size="2x" spin />
+      </div>
+    ) : exactMatches.length < 1 ? (
+      <div className="no-search-results-box">
+        We didnt find any <b>exact</b> matches based on search criteria.
```
Comment (kelfink) [inline app/javascript/views/searchUserList/SearchUserList.jsx]:
```suggestion
        We didn't find any <b>exact</b> matches based on search criteria.
```

File: app/javascript/_utils/formatters.test.js [hunk 3]
```diff
@@ -221,3 +222,40 @@ describe('#formatAdminWithRole', () => {
...
+  it('return empty when there is no role ', () => {
+    expect(formatOffices('', list)).toEqual('')
+  })
+
+  it('return original-id when it is not in the list ', () => {
+    expect(formatOffices(['bad'], list)).toEqual(['bad'])
+  })
+
+  it('return empty array when role is empty array', () => {
+    expect(formatOffices({}, list)).toEqual({})
```
Comment (Sushmareddy-OSI) [inline app/javascript/_utils/formatters.test.js]: Here, it's not returning the empty array. Can you change in the description ?

File: app/javascript/_utils/formatters.js [hunk 4]
```diff
@@ -59,6 +59,22 @@ export function formatRoles(assignedRoles, rolesList) {
...
+  const officeNames = [assignedOffices]
+  let office = ''
+  if (!Array.isArray(officeNames)) return ''
+  if (officeNames && officeNames.length !== 0) {
+    office = officeNames
+      .map(office => officesList.find(d => d.value === office))
+      .filter(value => value)
+      .map(({ value, label }) => label)[0]
+  }
+  if (office === undefined) {
```
Comment (kelfink) [inline app/javascript/_utils/formatters.js]: this block can be moved to after line 70, as it cannot be undefined unless lines 67-70 set the value to undefined.

File: app/javascript/_utils/formatters.test.js [hunk 5]
```diff
@@ -221,3 +222,40 @@ describe('#formatAdminWithRole', () => {
...
+    { value: 'bar', label: 'BAR_DESC' },
+    { value: 'quo', label: 'QUO_DESC' },
+    { value: 'qux', label: 'QUX_DESC' },
+  ]
+  it('return first index of the array ', () => {
+    expect(formatOffices('foo', list)).toEqual('FOO_DESC')
+    expect(formatOffices('qux', list)).toEqual('QUX_DESC')
+  })
+
+  it('return empty when there is no role ', () => {
```
Comment (kelfink) [inline app/javascript/_utils/formatters.test.js]:
```suggestion
  it('return empty when there is no office ', () => {
```

File: app/javascript/_utils/formatters.js [hunk 6]
```diff
@@ -59,6 +59,22 @@ export function formatRoles(assignedRoles, rolesList) {
   return role
 }
 
+export function formatOffices(assignedOffices, officesList) {
+  const officeNames = [assignedOffices]
```
Comment (kelfink) [inline app/javascript/_utils/formatters.js]: always wrapping the passed values with an array?  I think this implies that we're always one office or empty. Maybe we should just call this variable 'assignedOffice'. ??

Comment (RamuKammagani-osi) [reply app/javascript/_utils/formatters.js]: Updated in commit a89edd0

**Comment by RamuKammagani-osi** [comment]
> Please address comments made. Looks close to completed

@kelfink All the changes are addressed in the a89edd0

File: app/javascript/_utils/formatters.test.js (hunk 3 above)
Comment (RamuKammagani-osi) [reply app/javascript/_utils/formatters.test.js]: Updated in a89edd0

File: app/javascript/_utils/formatters.test.js [hunk 7]
```diff
@@ -242,12 +242,4 @@ describe('#formatOffices', () => {
   it('return original-id when it is not in the list ', () => {
     expect(formatOffices(['bad'], list)).toEqual(['bad'])
   })
-
-  it('return empty object when role is empty array', () => {
```
Comment (kelfink) [inline app/javascript/_utils/formatters.test.js]: offices ! ;-)

File: app/javascript/_utils/formatters.test.js [hunk 8]
```diff
@@ -242,12 +242,4 @@ describe('#formatOffices', () => {
...
-    expect(formatOffices({}, list)).toEqual({})
-  })
-
-  it('return empty array when role is empty ', () => {
```
Comment (kelfink) [inline app/javascript/_utils/formatters.test.js]: offices!

Comment (RamuKammagani-osi) [reply app/javascript/_utils/formatters.test.js]: @kelfink I have Updated it, You might have seen the old changes.

File: app/javascript/_utils/formatters.test.js (hunk 7 above)
Comment (RamuKammagani-osi) [reply app/javascript/_utils/formatters.test.js]: @kelfink I have Updated it, You might have seen the old changes.

**Pull request merged**